web/hooks/use-sign-in-redirection.ts

```diff
@@ -47,20 +48,26 @@ const useSignInRedirection = (): UseSignInRedirectionProps => {
 
         // Fetch the current user settings
         const userSettings: IUserSettings = await fetchCurrentUserSettings();
+        const workspacesList: IWorkspace[] = await fetchWorkspaces();
 
         // Extract workspace details
         const workspaceSlug =
           userSettings?.workspace?.last_workspace_slug || userSettings?.workspace?.fallback_workspace_slug;
 
         // Redirect based on workspace details or to profile if not available
-        if (workspaceSlug) router.push(`/${workspaceSlug}`);
+        if (
```

Comment:
@anmolsinghbhatia could you check if this same flicker wont occur if the url already has next path?